frappe/commands/site.py

```diff
@@ -1338,6 +1338,47 @@ def get_roles(context):
     if not context.sites:
         raise SiteNotSpecifiedError
 
+def add_new_user(email, first_name=None, last_name=None, send_welcome_email=False, password=None,role=None):
+    # add user
+    user = frappe.new_doc("User")
+    user.update(
+        {
+            "name": email,
+            "email": email,
+            "enabled": 1,
+            "first_name": first_name or email,
+            "last_name": last_name,
+            "user_type": "System User",
+            "send_welcome_email": 1 if send_welcome_email else 0,
+        }
+    )
+    user.insert()
+    user.add_roles(*role)
+    if password:
+        from frappe.utils.password import update_password
+        update_password(user=user.name, pwd=password)
+
+
+@click.command("list-roles")
+@pass_context
+def get_roles(context):
+    "get site available roles for users"
+    import frappe.utils.user
```

Comment:
You can easily add these commands very easily in your app instead: https://frappeframework.com/docs/v13/user/en/bench/extending-the-cli (basically copy the code you wrote to the commands file)


---

Also check out https://github.com/frappe/press/ 

FC also has a public API if you want to use it as a "backend" for your saas ops.


---

If you still want to add to the framework then here's review:

1. I don't see much use of commands like "get roles" in the framework. This is a very specific requirement for whatever you're trying to do. 
2. Creating user command is fine I guess. 


Follow this checklist: https://github.com/frappe/erpnext/wiki/Pull-Request-Checklist 

- [ ] docs
- [ ] formatting / import cleanup
- [ ] Add at last 1 test for this command in test_commands.py

Reply:
Thank you @ankush for your kind reply
I want to mention that getting roles will help us to know the roles that exist on site which will be different based on the apps installed on this site.
this is a helpful tool for anyone who needs to create a user but he does not know the fine role of it
And it helpful tool for any management tool trying to show the right roles for the one who is creating the user we still can add it to bench manager but it is really complementary for the creation user

let me know if it is OK to keep it on the boat

Reply:
Based on the conversation here with @gavindsouza no need for list-roles command
we remove it and we add a unit test for the command
and we fixing the formation issues
we are working on the doc